### PR TITLE
chore(rum-core): add native URL parser tests aganist polyfill

### LINKED ISSUE #534: Provide tests to check url.js polyfill against the native URL API
The url.js polyfill should be compatible with the native URL as much as possible. 
we should add tests that checks url.js outputs against the native API. 

In those case that we differ from the native API we should either fix our polyfill or document the difference in our tests.

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ fix elastic/apm-agent-rum-js#534 
+ Added incompatible tests in separate level to identify if we can align with spec closer in future.

### CODE REVIEW

File: packages/rum-core/test/common/url.spec.js [hunk 1]
```diff
@@ -269,3 +269,73 @@ describe('Url parser', function() {
     )
   })
 })
+
+/**
+ * In IE 11 - window.URL is not supported but it exists as an object
+ * Run these tests only on supported platforms
+ */
+if (window.URL.toString().indexOf('native code') !== -1) {
```
Comment (vigneshshanmugam) [inline packages/rum-core/test/common/url.spec.js]: We might have to skip this run on IE and Edge. Probably we can run only on chrome and ignore others.

Comment (vigneshshanmugam) [reply packages/rum-core/test/common/url.spec.js]: Ignored the test right now on IE and Edge. Test run on rest of the browsers.

**Comment by vigneshshanmugam** [comment]
@jahtalab In IE we cant do anything since its not supported. So we would have to skip running from there. And also we don't need to run on all browsers to know how we are differing from SPEC. 

For Edge, yes we can have that scenario separately for IPV6 address.

**Comment by jahtalab** [comment]
@vigneshshanmugam , your own tests show that we DO need to test on all browsers to see if they are different from the spec 😉

**Comment by vigneshshanmugam** [comment]
All of thee code review comments are addressed and the tests are adjusted to run on Microsoft Edge as well.

**Pull request merged**